### PR TITLE
Discard IPv6 header when using raw socket

### PR DESCRIPTION
**Pull request by mateuszkobak**
See RFC 3542

### CODE REVIEW

File: src/core/raw.c [hunk 1]
```diff
@@ -174,6 +174,15 @@ raw_input(struct pbuf *p, struct netif *inp)
         void *old_payload = p->payload;
 #endif
         ret = RAW_INPUT_DELIVERED;
+
+#if LWIP_IPV6
+        u8_t iphdr_version;
+        iphdr_version = IP_HDR_GET_VERSION(p->payload);
+        if (iphdr_version == 6) {
+          pbuf_remove_header(p, IP6_HLEN);
```
Comment (nalajcie) [inline src/core/raw.c]: I'm not sure this is fully-valid as you're skipping only the ipv6 main header, not all of the optional headers (see big while / switch before entering this function: https://github.com/phoenix-rtos/lwip/blob/STABLE-2_1_x_phoenix/src/core/ipv6/ip6.c#L1053). The RFC says that we should remove also all extension headers 
 (*the application will only receive the payload i.e., the data after the  IPv6 header and all the extension headers*)

Also, add some comments - eg. that it's a temporary workaround (we're stripping headers at a wrong place - before we know whether the user user `recv` or `recvmsg` interface - so this change makes it impossible to implement `recvmsg` correctly). Or try to do it in a correct place (proabbly tricker  :/).

Please update also the commit description, something along the lines:

RFCxxxx says that `recvfrom` should return only IPv6 packet payload without IPv6 and extension headers. 

To receive IPv6 headers `recvmsg` interface should be used (not implemented).

Comment (mateuszkobak) [reply src/core/raw.c]: Fixed this. Now we're skipping all the extension headers as well.

File: src/api/sockets.c [hunk 2]
```diff
@@ -1763,7 +1763,20 @@ lwip_sendto(int s, const void *data, size_t size, int flags,
     err = ERR_OK;
   }
 #else /* LWIP_NETIF_TX_SINGLE_PBUF */
-  err = netbuf_ref(&buf, data, short_size);
+  if (IP_IS_V6_VAL(buf.addr) && sock->conn->type == NETCONN_RAW_IPV6
```
Comment (nalajcie) [inline src/api/sockets.c]: I don't see any information regarding this error in the upstream, maybe it's worth to report it (with some minimal reproduction example) - they might be able to solve it in a better/more optimal way.

for now I suggest:
 - putting it under `#if LWIP_IPV6`
 - using correct indentation (2 spaces for if body,  4 spaces for multiline if)
 - we're missing `LWIP_CHECKSUM_ON_COPY` functionaligy`, maybe we should reuse the if body from the above, something along the lines:

```c
int needs_data_copy = 0;
#if LWIP_NETIF_TX_SINGLE_PBUF
needs_data_copy = 1;
#endif
#if LWIP_IPV6
if (IP_IS_V6_VAL(buf.addr) && (sock->conn->type == NETCONN_RAW_IPV6)
    && (sock->conn->pcb.raw->chksum_reqd == 1)) {
  /* [comment] */
  needs_data_copy = 1;
}

if (needs_data_copy) {
 // old allocation and other stuff
}
```

**Review by nalajcie — COMMENTED** [review]
I wold like to put it under `#if LWIP_IPV6` for performance reasons, feel free to apply other comments (or not ;))

File: src/api/sockets.c (hunk 2 above)
Comment (mateuszkobak) [reply src/api/sockets.c]: Ok, done

**Review by nalajcie — APPROVED** [review]

**Comment by agkaminski** [comment]
@mateuszkobak can this be merged?

**Review by anglov — APPROVED** [review]

**Pull request merged**